platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -241,22 +251,31 @@ def initialize_single_sfp(self, index):
             
             if not self._sfp_list[index]:
                 from .sfp import SFP
-                self._sfp_list[index] = SFP(index)
+                if self.RJ45_port_list and index in self.RJ45_port_list:
+                    self._sfp_list[index] = RJ45Port(index)
+                else:
+                    self._sfp_list[index] = SFP(index)
                 self.sfp_initialized_count += 1
 
     def initialize_sfp(self):
         if not self._sfp_list:
             from .sfp import SFP
             sfp_count = self.get_num_sfps()
             for index in range(sfp_count):
-                sfp_module = SFP(index)
+                if self.RJ45_port_list and index in self.RJ45_port_list:
+                    sfp_module = RJ45Port(index) #SFP(index, RJ45_TYPE)
```

Comment:
please remove unused code. if this should be a comment, have it on top of the code line